pkg/certificateprovider/azurekeyvault/provider.go

```diff
@@ -207,30 +210,42 @@ func parseAzureEnvironment(cloudName string) (*azure.Environment, error) {
 	return &env, err
 }
 
-func initializeKvClient(ctx context.Context, keyVaultEndpoint, tenantID, clientID string) (*kv.BaseClient, error) {
-	kvClient := kv.New()
+func initializeKvClient(ctx context.Context, keyVaultEndpoint, tenantID, clientID string, credProvider azcore.TokenCredential) (*azsecrets.Client, error) {
```

Comment:
correct me if I'm wrong, but I didn't see any caller pass in a non-nil credprovider.

Reply:
That's correct. It only is non-nill in unit tests for mocking purposes.

Reply:
I see. I feel it's not the best way to add this nil credProvider for testing purpose, but may need some refactoring on how we invoke `initKVClient` function. It's out of the scope of this PR, could you add some TODO comments so that we can address it later.